### PR TITLE
Fixes for main

### PR DESCRIPTION
**Pull request by beckertobias**
SignIn should work now ✅ 

Regarding SignUp:
- I added error handling so we don't have an undhandled error anymore
- But we are still throwing an ApolloError because:

The thing is that I have the following line in my code: `const user = await getTalentById(uid);`. And I don't know in advance if I have a user with that uid or not. So I want to return the user that I get (either `undefined` or an `user`). 

But in the new version of main we do `throw new ApolloError` in the helper function `getTalentById` if we didn't find a user with an ID / UID.

Suggestion: Let's not throw an error in the helper function and instead return talent or undefined.

### CODE REVIEW

**Comment by atsteinberg** [comment]
Sure. Change the graphql if that is necessary for our signup to work. Could you do it for this pr? Or do you want to create a new one?

**Comment by beckertobias** [comment]
Implemented it now as suggested.
Due to the bad internet connection in the train, I fail on testing the signup (just get an error because server cannot be reached - same for graphql playground). 
Need to test that properly before merging.

**Comment by beckertobias** [comment]
@atsteinberg  @camifernweh  @pintoedo signup is working now. 
PLease somebody approve. Then main is working at least and we can continue with all the other TO DOs

**Review by atsteinberg — COMMENTED** [review]
Hm. I'm not sure I like these changes. From the standpoint of graphql, asking for a talent whose uid does not exist in the db *should* throw an error. Why is it impossible to expect it at this point and deal with it?

My problem is: I've tried to figure out which line actually makes it the case that the 400 response error is logged to the console. Can you help me and walk me through it one of these days? Sorry for being difficult, but there should be an easy way of treating server errors without them showing in the console, or am I missing something here?

File: apollo/helpers.ts
Comment (TobiBecker) [inline apollo/helpers.ts]: @atsteinberg this line was throwing the error, if I recall correctly

**Comment by atsteinberg** [comment]
yes. Sure. For my taste that is fine for the error to be thrown there.
But why does the error show up in the browser console? I agree that it shouldn’t be there (as a console log to the browser console), since we expect to trigger the graphql error in this case.

**Comment by TobiBecker** [comment]
> Hm. I'm not sure I like these changes. From the standpoint of graphql, asking for a talent whose uid does not exist in the db _should_ throw an error. Why is it impossible to expect it at this point and deal with it?
> 

I am dealing with the error. This is not the issue. The issue is that I don't want an error to be thrown each time somebody signs up 😅 
But the signup is working - even with the error.

If you want GraphQL to throw an error in this case, we have to make it different. Then we need one query to check if there is a talent with an uid. And if there is one, we call another query that gets us the query. Not sure if there is a better solution.

**Comment by atsteinberg** [comment]
Again, my issue is why the error message shows up in the browser console. Do you know?

**Comment by TobiBecker** [comment]
ah, just saw your other comment. I can answer later, I'm in a meeting. But I'm not sure why it shows up in the console. At least not out of my head

**Comment by TobiBecker** [comment]
yeah sorry, my comment was delayed. Didn't see your other comment when I submitted

**Comment by TobiBecker** [comment]
looked into it. But couldn't figure it out yet.
The `throw new ApolloError` makes the error appear in the console of the server - which is fine.
Screenshot 2021-02-25 at 17 01 16

But at the same time we receive a 400 in the console. Maybe we have to handle the case for 'no talent found' better in the helper. Meaning that if we don't find a talent, we set the status code to something and return sth.

400 in the browser console:
Screenshot 2021-02-25 at 17 01 05

**Comment by atsteinberg** [comment]
That is worth a shot. Although, is there a more suitable status code in this case than 400? In any case, I think this:
https://www.apollographql.com/docs/react/data/error-handling/ <https://www.apollographql.com/docs/react/data/error-handling/>
Is the place to look (especially the very last bit).
Anyway: so it seems that Apollo client is responsible for logging the error to the console. So, we somehow need to tell it to not do that in this particular case and we’re all set.
Do you agree?

**Comment by atsteinberg** [comment]
I was very confused about this, since the getTalentById query was executed in a try-catch block. And even when leaving the catch block empty it still gets thrown. So I had no idea what was going on. But makes sense if it part of Apollo client’s default error handling to log to the (browser) console.

**Comment by atsteinberg** [comment]
I think, the trick is to pass an options object to the useQuery with an onError key:
https://www.apollographql.com/docs/react/data/queries/#usequery-api <https://www.apollographql.com/docs/react/data/queries/#usequery-api>

If you have it open, you could just try to pass something like
{ onError:(error: ApolloError} => if (!error) console.log(’no error’) }
As options object to the relevant useQuery (shouldn’t do anything and typescript and linters don’t complain.

**Comment by beckertobias** [comment]
Hi Alex, it is working now.
What I did:
- added {loading, error} to the mutation options --> no error logged to browser console
- reverted the changes to getTalentById, i.e. it has to return a Talent
- checked signup again --> no error logged to browser console
- deleted the {loading, error} option --> still no error logged to console 😅 

So it is working as we want it to work. I am only not quite sure why we don't get the error in the browser console now 

Please approve if it is fine for you now.

**Comment by atsteinberg** [comment]
Ok. Will confirm that I get the same behaviour.

File: apollo/typedefs.ts
Comment (atsteinberg) [inline apollo/typedefs.ts]: could you change this back to return Talent!
?

File: lib/auth/firebaseAdmin.ts
Comment (atsteinberg) [inline lib/auth/firebaseAdmin.ts:38]: could you add eslint-ignore-line no-console (I think that's the right command) to this and all other console logs that will stay in the code for the time being, and remove all logs that only are there for checking purposes?

File: pages/signup/index.tsx
Comment (atsteinberg) [inline pages/signup/index.tsx:34]: I have the required changes on my computer. Will push directly to main once this pr is merged, if that's ok.

File: apollo/typedefs.ts
Comment (beckertobias) [reply apollo/typedefs.ts]: done

File: lib/auth/firebaseAdmin.ts
Comment (beckertobias) [reply lib/auth/firebaseAdmin.ts:38]: done

File: pages/signup/index.tsx
Comment (beckertobias) [reply pages/signup/index.tsx:34]: ok, fine for me.

**Comment by beckertobias** [comment]
@atsteinberg did requested changes

**Review by atsteinberg — APPROVED** [review]
🕺

**Pull request merged**
